packages/lib/passport-interface/src/Credential.ts

```diff
@@ -152,15 +154,21 @@ export async function verifyCredential(
         return {
           email: emailPCD.claim.emailAddress,
           signer: emailPCD.proof.eddsaPCD.claim.publicKey,
-          semaphoreId: emailPCD.claim.semaphoreId
+          semaphoreId: emailPCD.claim.semaphoreId,
+          semaphoreV4Id: emailPCD.claim.semaphoreV4Id
         } satisfies SignedEmail;
       })
     );
 
+    // a bit hacky
+    const semaphoreV4Id = signedEmails.find((e) => e.semaphoreV4Id)
+      ?.semaphoreV4Id;
+
```

Comment:
not really sure we even need this, i'm going to keep track of it in my head prior to devcon and see if we need to evolve it.

Reply:
not sure if we need what exactly?

Reply:
the thing i'm doing here is extracting a semaphore v4 identity from the email pcd and setting that as the user's single v4 identity. this is sort of wrong in the sense that the user could technically upload multiple email pcds with multiple semaphore v4 identities, but in practice this won't happen. furthermore this semaphore v4 id field in the `VerifiedCredential` is not something that is really used anywhere, and if it is we could get that value a different way. tl;dr this is an implementation detail that's not super relevant for you to track but i'm going to remember it and make sure something right is done with it in the future.